### PR TITLE
Change `get_current_traceback` to `DebugTraceback`

### LINKED ISSUE #1992: ImportError: cannot import name 'get_current_traceback' from 'werkzeug.debug.tbtools'
I am working on a Dash app and it was working great till yesterday. When I tried today, I am seeing the following:

Traceback (most recent call last):
  File "/app/app-v2.py", line 12, in <module>
    import dash
  File "/usr/local/lib/python3.9/site-packages/dash/__init__.py", line 5, in <module>
    from .dash import Dash, no_update  # noqa: F401,E402
  File "/usr/local/lib/python3.9/site-packages/dash/dash.py", line 18, in <module>
    from werkzeug.debug.tbtools import get_current_traceback
ImportError: cannot import name 'get_current_traceback' from 'werkzeug.debug.tbtools' (/usr/local/lib/python3.9/site-packages/werkzeug/debug/tbtools.py)

I am running the app on macOS Catalina. Happy to provide more details.

### PR DESCRIPTION
**Pull request by deepyaman**
Closes #1992 

It's not an option to change `get_current_traceback` to `DebugTraceback` _entirely_ because `Werkzeug==2.1.0` requires Python 3.7, but we can backport a minimal `DebugTraceback` to earlier versions of Werkzeug.

## Contributor Checklist

- [ ] I have broken down my PR scope into the following TODO tasks
   -  [ ] task 1
   -  [ ] task 2
- [ ] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [ ] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [x] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follows
    -  [ ] this GitHub [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in Plotly Dash community

### CODE REVIEW

**Comment by deepyaman** [comment]
Closing this PR, as upgrading to Werkzeug 2.1.0 would mean dropping support for Python 3.6, which is a much bigger decision.

File: requires-install.txt
Comment (deepyaman) [inline requires-install.txt:2]:
```suggestion
Werkzeug<2.1.0  # Werkzeug 2.1.0 dropped support for Python 3.6
click<8.1.0  # click 8.1.0 dropped support for Python 3.6
```

File: requires-all.txt
Comment (deepyaman) [inline requires-all.txt:34]:
```suggestion
Werkzeug<2.1.0  # Werkzeug 2.1.0 dropped support for Python 3.6
click<8.1.0  # click 8.1.0 dropped support for Python 3.6
```

File: CHANGELOG.md
Comment (deepyaman) [inline CHANGELOG.md:10]:
```suggestion
- [#1993](https://github.com/plotly/dash/pull/1993) Pin Flask<2.1.0, Werkzeug<2.1.0, and click<8.1.0 to retain Python 3.6 compatibility, resolving [#1992](https://github.com/plotly/dash/issues/1992).
```

**Review by flying-sheep — CHANGES_REQUESTED** [review]
Version upper bounds are no good approach for library projects: https://iscinumpy.dev/post/bound-version-constraints/

You should instead write conditional code handling all flask/werkzeug versions or (if that’s too hard) drop Python 3.6 support

**Comment by deepyaman** [comment]
> Version upper bounds are no good approach for library projects: https://iscinumpy.dev/post/bound-version-constraints/
> 
> You should instead write conditional code handling all flask/werkzeug versions or (if that’s too hard) drop Python 3.6 support

@flying-sheep I can do either. I don't see much examples of conditional Python version handling within Dash (only https://github.com/plotly/dash/blob/3334c07f59cb1ee9bc4596ebbdce6edb41f94968/dash/development/component_generator.py#L154 outside of tests, but even that's for a component generator), but I've seen plenty of it in other libraries. I would personally also be fine to drop 3.6, but don't think that's my call.

Are you one of the maintainers of Dash? Just want to makes sure the change is aligned with whoever will be making the final call.

**Comment by flying-sheep** [comment]
I’m not, sorry if I gave the impression. I think you can see that by the color of my review icon: mine is gray, so it’s just a “suggested change” and not a “requested change”.

I’m just speaking from my history of dealing with compatibility problems and bugs in python libraries. That blog post I linked really nails it IMHO, and I think any library maintainer should read it.

**Comment by alexcjohnson** [comment]
Py3.6 still accounts for about [10% of our downloads](https://pypistats.org/packages/dash) so I don't think it's time to drop it yet. 3.7 is still the most used though it's about to lose that title to 3.8 finally.

File: requires-all.txt
Comment (deepyaman) [inline requires-all.txt:5]:
```suggestion
black==21.6b0;python_version<"3.7"
black==22.3.0;python_version>="3.7"
```

File: requires-ci.txt
Comment (deepyaman) [inline requires-ci.txt:2]:
```suggestion
black==21.6b0;python_version<"3.7"
black==22.3.0;python_version>="3.7"
```

**Comment by alexcjohnson** [comment]
Thanks @deepyaman - we ended up going with #1995 so I'll close this one but your work here was helpful toward fully understanding the issue!

**Pull request closed without merging**
